distributed/tests/test_steal.py

```diff
@@ -1350,3 +1350,36 @@ def test_steal_worker_state(ws_with_running_task):
     assert "x" not in ws.tasks
     assert "x" not in ws.data
     assert ws.available_resources == {"R": 1}
+
+
+@pytest.mark.slow()
+@gen_cluster(nthreads=[("", 1)] * 4, client=True)
+async def test_steal_very_fast_tasks(c, s, *workers):
+    # Ensure that very fast tasks are allowed to be stolen
+    root = dask.delayed(lambda n: "x" * n)(
+        dask.utils.parse_bytes("1MiB"), dask_key_name="root"
+    )
+
+    @dask.delayed
+    def func(*args):
+        import time
+
+        time.sleep(0.002)
+
+    ntasks = 1000
+    results = [func(root, i) for i in range(ntasks)]
+    futs = c.compute(results)
+    await c.gather(futs)
+
+    # The following is an attempt to estimate and assert a roughly uniform data
+    # distribution
+    max_ntasks_on_worker = 0
+    for w in workers:
+        ntasks_on_worker = len(w.data)
+        if ntasks_on_worker > max_ntasks_on_worker:
+            max_ntasks_on_worker = ntasks_on_worker
+        assert ntasks_on_worker > ntasks / len(workers) * 0.5
+
+    ntasks_on_other_workers = ntasks - max_ntasks_on_worker
+    nother_workers = len(workers) - 1
+    assert max_ntasks_on_worker < 2 * ntasks_on_other_workers / nother_workers
```

Comment:
```suggestion
    ntasks_per_worker = np.array([len(w.data) for w in workers])
    ideal = ntasks / len(workers)
    assert (ntasks_per_worker > ideal * 0.5).all(), (ideal, ntasks_per_worker)
    assert (ntasks_per_worker < ideal * 1.5).all(), (ideal, ntasks_per_worker)
```
To me this math is a bit hard to read. Could you just do something like this? I know the check for the max I have here is different from what you have, but I'm not sure it needs to be as complicated as what you have?